test/bpf-loader.test.js

```diff
@@ -51,18 +51,21 @@ test('load BPF Rust program', async () => {
   const data = await fs.readFile(
     'test/fixtures/noop-rust/solana_bpf_rust_noop.so',
   );
-
   const connection = new Connection(url);
   const [, feeCalculator] = await connection.getRecentBlockhash();
   const fees =
     feeCalculator.lamportsPerSignature *
-    (BpfLoader.getMinNumSignatures(data.length) + NUM_RETRIES);
-  const from = await newAccountWithLamports(connection, fees);
+    (BpfLoader.getMinNumSignatures(data.length) + NUM_RETRIES + 1);
+  const payer = await newAccountWithLamports(connection, fees);
+  const programId = await BpfLoader.load(connection, payer, data);
+  const program_data = Buffer.from([1, 2, 3, 4, 5, 6, 7, 8]);
 
-  const programId = await BpfLoader.load(connection, from, data);
-  const transaction = new Transaction().add({
-    keys: [{pubkey: from.publicKey, isSigner: true, isDebitable: true}],
-    programId,
-  });
-  await sendAndConfirmTransaction(connection, transaction, from);
+  const transaction = new Transaction().add(
+    Loader.invokeMainInstruction({
```

Comment:
I don’t think I grok what you’re doing here.  Where does the 1,2,3,4,5,6,7,8 input go exactly? The programId is of the rust_noop.so and not the BPFLoader right?

Reply:
Yes, the programId returned from load is rust_noop.so's and the `programData` is the instruction data sent to noop's process instruction

Reply:
`Loader` encapsulates `LoaderInstruction` (Write, Finalize, InvokeMain).  Yes, the examples change so they don't have to construct the `InvokeMain` variant of `LoaderInstruction` themselves.  We could put `invokeMainInstruction` someplace else but either way, the examples change.  This PR coincide with https://github.com/solana-labs/solana/pull/6470 where the runtime no longer assumes `InvokeMain`.   This allows loaders to extend or replace `LoaderInstruction` at the client/loader level without forcing a Solana runtime change.

Reply:
Here is an example of how the examples invoke a program: https://github.com/solana-labs/example-messagefeed/pull/201